src/daemons/rhsmcertd.c

```diff
@@ -504,6 +555,31 @@ key_file_init_config (Config * config, GKeyFile * key_file)
             DEFAULT_AUTO_REGISTRATION
             );
     config->auto_registration = auto_registration_enabled;
+
+    gchar *default_log_level = get_string_from_config_file(
+            key_file,
+            "logging",
+            "default_log_level"
+            );
+
+    gchar *rhsmcertd_log_level = get_string_from_config_file(
+            key_file,
+            "logging",
+            "rhsmcertd"
+            );
+
+    if (rhsmcertd_log_level == NULL && default_log_level != NULL) {
+        set_log_level(default_log_level, "logging.default_log_level");
+    } else if (rhsmcertd_log_level != NULL) {
+        set_log_level(rhsmcertd_log_level, "logging.rhsmcertd");
+    }
+
+    if (default_log_level != NULL) {
+        g_free(default_log_level);
+    }
+    if (rhsmcertd_log_level != NULL) {
+        g_free(rhsmcertd_log_level);
+    }
```

Comment:
`g_free()` is a wrapper around `free()`, and `free(NULL)` is defined by POSIX to be a no-op; hence, please remove the if != null checks, as they are effectively redundant